add Article model, migration and seed

NoahRGuy opened the pull request.

Commit:
add Article model, migration and seed

TylerSmall19 commented on db/migrate/20170105160943_create_articles.rb:
```diff
@@ -0,0 +1,10 @@
+    create_table :articles do |t|
+      t.string :title, null: false
+      t.text :body
+      t.integer :author_id
```
Body and author_id should be `null: false`, right?

TylerSmall19 commented on app/models/article.rb:
```diff
@@ -0,0 +1,8 @@
+  has_many :categories, through: :categorizations
+  has_many :revisions
+
+  validates :title, presence: true, uniqueness: true
```
Body and Author_id should be required?

Commit:
Tweaks to code to fix for PR request (Articles)

https://github.com/TylerSmall19/cotton-tales/pull/13

Added not null validations to Article migration file
delete ghost directory 'rabbit-site'
Added validation for Article model in regards to author id

Commit:
commenting out current work so PR passes tests to merge
